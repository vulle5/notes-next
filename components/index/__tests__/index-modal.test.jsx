import React from 'react'
import { render } from 'testing/testing-library-wrapper'
import * as nextRouter from 'next/router'

import IndexModal from '../index-modal'

beforeEach(() => {
  const modalRoot = document.createElement('div')
  modalRoot.setAttribute('id', '__next')
  document.body.appendChild(modalRoot)
})

const mockUseRouter = returnValue => {
  nextRouter.useRouter = jest.fn().mockImplementation(() => returnValue)
}

test('should render correctly', () => {
  mockUseRouter({ query: {} })
  render(<IndexModal />)
  expect(document.body).toMatchSnapshot()
})

test('should render modal when noteId in query', () => {
  mockUseRouter({ query: { noteId: 'some-id' } })
  const { getByRole } = render(<IndexModal />)
  expect(getByRole('dialog')).toMatchSnapshot()
})
