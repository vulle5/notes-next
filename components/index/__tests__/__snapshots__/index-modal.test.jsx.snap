// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<body>
  <div
    id="__next"
  />
  <div />
  <div
    class="ReactModalPortal"
  />
</body>
`;

exports[`should render modal when noteId in query 1`] = `
<div
  class="ReactModal__Content ReactModal__Content--after-open"
  id="modal-content"
  role="dialog"
  style="position: relative; top: 40%; left: 50%; border: transparent; background: none; overflow: auto; border-radius: 4px; outline: none; padding: 0px; transform: translate(-50%, -50%) scale(1); opacity: 1; max-width: 500px; transition: transform .4s cubic-bezier(0.15, 1, 0.75, 1), opacity .4s;"
  tabindex="-1"
>
  <div
    class="shadow-none"
    style="padding: 1rem;"
  >
    <div
      class="jsx-2455388080 close-button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-times-circle fa-w-16 fa-lg "
        data-icon="times-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
          fill="currentColor"
        />
      </svg>
    </div>
    <form
      class="jsx-398696813 wrapper"
    >
      <input
        class="jsx-398696813 title-input"
        id="note-title"
        name="note-title"
        placeholder="Title"
        type="text"
        value=""
      />
      <textarea
        class="jsx-1612382694 "
        id="note-content"
        name="note-content"
        placeholder="Remind me to..."
        rows="1"
        type="text"
      />
      <button
        class="filled"
        style="width: auto;"
        type="submit"
      >
        <div
          class="buttonContent"
        >
          Add proper title
        </div>
      </button>
    </form>
  </div>
</div>
`;
